models/tofu/polygon/tofu_polygon_events.sql

```diff
@@ -0,0 +1,150 @@
+{{ config(
+    alias = 'events',
+    partition_by = ['block_date'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash', 'evt_index', 'bundle_index' ],
+    post_hook='{{ expose_spells(\'["polygon"]\',
+                                "project",
+                                "tofu",
+                                \'["Henrystats","sohwak"]\') }}')
+}}
+
+{% set MATIC_ADDRESS = '0x0000000000000000000000000000000000001010' %}
+{% set project_start_date = '2021-11-01' %}
+
+WITH tff AS (
+    SELECT call_block_time,
+           call_tx_hash,
+           id,
+           fee_rate,
+           royalty_rate,
+           fee_address,
+           royalty_address,
+           bundle_size,
+           get_json_object(t, '$.token')   as token,
+           get_json_object(t, '$.tokenId') as token_id,
+           get_json_object(t, '$.amount')  as amount,
+           i as bundle_index
+    FROM (SELECT call_block_time,
+                 call_tx_hash,
+                 get_json_object(detail, '$.id')                                                     as id,
+                 get_json_object(get_json_object(detail, '$.settlement'), '$.feeRate') / 1000000     as fee_rate,
+                 get_json_object(get_json_object(detail, '$.settlement'), '$.royaltyRate') / 1000000 as royalty_rate,
+                 get_json_object(get_json_object(detail, '$.settlement'), '$.feeAddress')            as fee_address,
+                 get_json_object(get_json_object(detail, '$.settlement'), '$.royaltyAddress')        as royalty_address,
+                 posexplode(from_json(get_json_object(detail, '$.bundle'), 'array<string>'))         as (i,t),
+                 json_array_length(get_json_object(detail, '$.bundle'))                              as bundle_size
+          FROM {{ source('tofu_nft_polygon', 'MarketNG_call_run') }}
+          WHERE call_success = true
+              {% if is_incremental() %}
+              and call_block_time >= date_trunc("day", now() - interval '1 week')
+              {% endif %}
+         ) as tmp
+),
+     tfe as (
+         select evt_tx_hash,
+                evt_block_time,
+                evt_block_number,
+                evt_index,
+                id,
+                get_json_object(inventory, '$.seller')   as seller,
+                get_json_object(inventory, '$.buyer')    as buyer,
+                get_json_object(inventory, '$.kind')     as kind,
+                get_json_object(inventory, '$.price')    as price,
+                CASE WHEN get_json_object(inventory, '$.currency') = '0x0000000000000000000000000000000000000000'
+                  THEN '{{MATIC_ADDRESS}}'
+                  ELSE get_json_object(inventory, '$.currency')
+                END as currency,
+                (get_json_object(inventory, '$.currency') = '0x0000000000000000000000000000000000000000') as native_eth,
```

Comment:
```suggestion
                (get_json_object(inventory, '$.currency') = '0x0000000000000000000000000000000000000000') as native_matic,
```

Let's change this for more clarity.